src/EditorFeatures/Core/Implementation/SmartIndent/IBlankLineIndentationService.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using Microsoft.CodeAnalysis.Formatting;
+
+namespace Microsoft.CodeAnalysis.Editor
+{
+    internal interface IBlankLineIndentationService
```

Comment:
💭 I'd rather see this included in `IIndentationService`. We're up to 3 interfaces for 1 concept, even though it's an internal interface for the purpose of allowing us to make changes to the API shape.

Reply:
I can't really change that interface though as this is something implemented by our partner teams.  I Agree the 3 interfaces for 1 concept isn't great.  In a similar fashion, we also have this really hairy split between the indenter and the formatter (which i believe shouldn't even be necessary now).  I mentioned in the OP that i'm trying to clean this up a small step at a time (especially to keep risk low).

Note: i could get rid of this interface, and just have people depend on the exposed class method... but that seems a bit skeezy to me.